### PR TITLE
emit appimage-filename-updated event

### PR DESCRIPTION
**Pull request by gaodeng**
The appimage-filename-updated event is emitted here, so that if the appimage filename is updated, the app will have the opportunity to update the path in the .desktop file

### CODE REVIEW

File: packages/app-builder-lib/src/electron/electronMac.ts [hunk 1]
```diff
@@ -121,7 +121,7 @@ export async function createMacApp(packager: MacPackager, appOutDir: string, asa
     helper.CFBundleDisplayName = `${appInfo.productName} Helper ${postfix}`
     helper.CFBundleIdentifier = userProvidedBundleIdentifier
       ? filterCFBundleIdentifier(userProvidedBundleIdentifier)
-      : `${helperBundleIdentifier}.${postfix.replace(/[^a-z0-9]/gim, "")}`
+      : filterCFBundleIdentifier(`${helperBundleIdentifier}.${postfix}`)
```
Comment (develar) [inline packages/app-builder-lib/src/electron/electronMac.ts:124]: What's the reason for this change?

Comment (gaodeng) [reply packages/app-builder-lib/src/electron/electronMac.ts:124]: https://github.com/electron-userland/electron-builder/pull/4154
In this pr, I didn't realize that this filterCFBundleIdentifier function existed at that time, now I cleaned it up and replaced it with calling this filterCFBundleIdentifier function to avoid duplicate code

Comment (gaodeng) [reply packages/app-builder-lib/src/electron/electronMac.ts:124]: @develar  any news?

**Comment by gaodeng** [comment]
Yes it's still relevant.

**Comment by gaodeng** [comment]
any news?

**Comment by gaodeng** [comment]
Yes it's still relevant.

**Comment by gaodeng** [comment]
Yes it's still relevant.
